back/engines/commercial/insights/app/models/insights/patches/project.rb

```diff
@@ -8,9 +8,16 @@ def self.included(base)
           has_many(
             :insights_data_sources,
             class_name: 'Insights::DataSource',
-            foreign_key: :origin_id,
-            dependent: :destroy
+            foreign_key: :origin_id
           )
+
+          after_destroy :destroy_insights_views
```

Comment:
Did you figure out why we need to use `after_destroy` hooks instead of `before` hooks?

Reply:
Not really, but rails doesn't seem to really support these cycles in dependent: destroy behavior, which we have here. Found others having issues with it.